model/model.js

```diff
@@ -6,20 +6,27 @@ exports.selectAllModels = () => {
 	});
 };
 
-exports.selectAllArticles = () => {
-	return db
-		.query(
-			`
-  SELECT a.article_id, a.title, a.topic, a.author, a.created_at, a.votes, a.article_img_url, 
+exports.selectAllArticles = (topic, sort_by = 'created_at', order = 'DESC') => {
+
+	let query = `SELECT a.article_id, a.title, a.topic, a.author, a.created_at, a.votes, a.article_img_url, 
   COUNT(c.article_id) AS comment_count 
   FROM articles a 
-  JOIN comments c ON a.article_id = c.article_id 
-  GROUP BY a.article_id 
-  ORDER BY a.created_at DESC;`
-		)
-		.then((articles) => {
-			return articles.rows;
-		});
+  JOIN comments c ON a.article_id = c.article_id `;
+
+	let queryValues = [];
+	if (topic) {
+		query += `WHERE a.topic = $1 `;
+		queryValues.push(topic);
+	}
+
+	query += `GROUP BY a.article_id ORDER BY ${sort_by} ${order};`;
+
+	return db.query(query, queryValues).then((articles) => {
+		if (articles.rows.length === 0) {
+			return Promise.reject({ status: 400, msg: 'Bad request' });
+		}
+		return articles.rows;
+	});
```

Comment:
Good stuff here, we need to ensure we are also protected against SQL injection. If the variables injected into the string are invalid we shouldn't pass these to the SQL query as we risk malicious lines being used here. Could we ensure we check for that before being passed into the query?